core/blockchain/impl/block_storage_impl.hpp

```diff
@@ -60,7 +60,7 @@ namespace kagome::blockchain {
     outcome::result<primitives::BlockHash> putBlockHeader(
         const primitives::BlockHeader &header) override;
 
-    outcome::result<std::optional<primitives::BlockHeader>> getBlockHeader(
+    outcome::result<primitives::BlockHeader> getBlockHeader(
```

Comment:
It should be the other way around, BlockHeaderRepository should return an optional header. However, it's hard to refactor it now since it's used in a mighty lot of places. So here where it's already done correctly changing it to the worse BlockHeaderRepository interface is not supported by me.

Reply:
In many (if not all) cases nullopt result was considered as an error but processed outside. It's much more simple to return block_not_found error right from this function.

Reply:
Added tryGetBlockHeader to block_storage so as unit tests for both get and tryGet methods